### ISSUE TITLE
cf apps fails with timeout

### ISSUE THREAD

**Issue by Infra-Red** [cloudfoundry/cli#855]
### Command

`cf apps` should return apps list, but instead i get dial tcp: i/o timeout

### CLI Version

```
cf -v
cf version 6.18.1+a1103f0-2016-05-24
```

### Error

When primary system DNS broken `cf a` command does not return a result and fails with 15s timeout:
```
time /usr/bin/cf a
Getting apps in org admin / space apps as admin...
FAILED
Error performing request: Get https://api.cf.example.corp/v2/spaces/baafe58e-7902-4e97-b1c8-a945c1f9d70a/summary: dial tcp: i/o timeout
TIP: If you are behind a firewall and require an HTTP proxy, verify the https_proxy environment variable is correctly set. Else, check your network connection.
real	0m15.108s
user	0m0.088s
sys	0m0.020s
```

At the same time i able to reach CF API using curl command:
```
curl -k https://api.cf.example.corp/v2/spaces/baafe58e-7902-4e97-b1c8-a945c1f9d70a/summary
{
  "code": 10002,
  "description": "Authentication error",
  "error_code": "CF-NotAuthenticated"
}
```

Maybe CF cli have some properties for DNS check timeout?

### CF Trace

```
CF_TRACE=true /usr/bin/cf a
Getting apps in org admin / space apps as admin...

REQUEST: [2016-06-08T07:50:17-07:00]
GET /v2/spaces/baafe58e-7902-4e97-b1c8-a945c1f9d70a/summary HTTP/1.1
Host: api.cf.example.corp
Accept: application/json
Authorization: [PRIVATE DATA HIDDEN]
Content-Type: application/json
User-Agent: go-cli 6.18.1+a1103f0 / linux


FAILED
Error performing request: Get https://api.cf.example.corp/v2/spaces/baafe58e-7902-4e97-b1c8-a945c1f9d70a/summary: dial tcp: i/o timeout
TIP: If you are behind a firewall and require an HTTP proxy, verify the https_proxy environment variable is correctly set. Else, check your network connection.
```

### CC API Endpoint Version

`2.48.0`

### Platform Details

Ubuntu 14.04.4 64-bit

### Shell

Terminal

**Comment by dkoper** [comment]
Hi @InfraRed,

Would you mind trying with our edge binary and report back here? We found an issue with the way the cf CLI was built on Linux 64 bit that may affect the DNS resolver used. A fix for this is expected in the next cf CLI release and is already incorporated in the edge binary.
https://github.com/cloudfoundry/cli#downloads
Link is under the Homebrew section.

Cheers,
Dies Koper
CF CLI PM

**Comment by Infra-Red** [comment]
I just checked and got the same result:
```
$ ~/cf a
Getting apps in org admin / space demo as admin...
FAILED
Error performing request: Get https://api.cf.example.corp/v2/spaces/77f7bf05-811c-4c3a-9c88-a78b15076dde/summary: dial tcp: i/o timeout
TIP: If you are behind a firewall and require an HTTP proxy, verify the https_proxy environment variable is correctly set. Else, check your network connection.
$  ~/cf -v
cf version 6.18.1+026f9b3-2016-06-08
```

**Comment by dkoper** [comment]
The issue is that the cf CLI has a (hardcoded) 5 sec timeout for resolving the domain name and connecting to the IP address. So if /etc/resolv.conf is configured to switch to the secondary DNS server after 5 seconds (its documented default timeout value), the CLI will already have given up.
In cf CLI 6.20.0 we have experimentally replaced the hardcoded value with an environment variable `CF_DIAL_TIMEOUT` to allow you to override the 5 sec timeout. Please let us know how that works out for you.

Cheers,
Dies Koper
CF CLI PM

**Comment by dsboulder** [comment]
@dkoper sounds like this issue was resolved by making the DNS timeout configurable.  Should we close the GH issue and the tracker story (https://www.pivotaltracker.com/story/show/123654531)?

**Comment by dkoper** [comment]
Thanks again for bringing this up. The `CF_DIAL_TIMEOUT` environment variable is now exposed in the help text.
https://github.com/cloudfoundry/cli#downloads

Cheers,
Dies Koper
CF CLI PM

**Comment by munilvc** [comment]
Thank you @dkoper for this solution.  The CF_DIAL_TIMEOUT fixed the problem on my side too.   However if was not clear for me to find the solution in the help options, could you please point the subject where this configuration is described?  for isntance: "cf help config" or similar ? Thanks again!

**Comment by dkoper** [comment]
@munilvc It is supposed to appear in the ENVIRONMENT VARIABLES section of `cf help -a`.
It seems it has disappeared.
Thanks for pointing it out, we'll put it back in in the next release:
https://www.pivotaltracker.com/story/show/131497359

**Comment by dkoper** [comment]
@munilvc FYI, it's back again in `cf help -a` in the just published 6.22.2 release. Thanks again for bringing up that it wasn't there anymore in the help options.

**Comment by munilvc** [comment]
This is great, thank you very much for the fix and the update!
